index.bs

```diff
@@ -4109,165 +4310,1257 @@ To <dfn>get the response data</dfn> given |response|:
+
+[=Remote end definition=]
+
+<pre class="cddl remote-cddl">
+network.UrlPattern = network.UrlPatternPattern // network.UrlPatternString
+
+network.UrlPatternPattern = {
+    type = "pattern",
+    ?protocol: text,
+    ?hostname, text,
+    ?port: text,
+    ?pathname: text,
+    ?search: text,
+}
+
+
+network.UrlPatternString = {
+    type = "string",
+    pattern: text,
+}
+
+</pre>
+
+A <code>network.UrlPattern</code> represents a pattern used for matching request
+URLs for [=network intercepts=].
+
+When URLs are matched against a <code>network.UrlPattern</code> the URL is
+parsed, and each component is comapred for equality with the corresponding field
+in the pattern, if present. Missing fields from the pattern always match.
+
+Note: This syntax is designed with future extensibility in mind. In particular
+the syntax forbids characters that are treated specially in the [[URLPattern]]
+specification. These can be escaped by prefixing them with a U+005C (\) character.
+
+<div algorithm>
+To <dfn>unescape URL pattern</dfn> given |pattern|
+
+1. Let |forbidden characters| be the [=/set=] of codepoints «U+0028 ((), U+0029
+   ()), U+002A (*), U+007B ({), U+007D (})»
+
+1. Let |result| be the empty string.
+
+1. Let |is escaped character| be false.
+
+1. For each |codepoint| in |pattern|:
+
+  1. If |is escaped character| is false:
+
+    1. If |forbidden characters| [=set/contains=] |codepoint|, return [=error=]
+       with [=error code=] [=invalid argument=].
+
+    1. If |codepoint| is U+005C (\):
+
+        1. Set |is escaped character| to true.
+
+        1. Continue.
+
+  1. Append |codepoint| to result.
+
+  1. Set |is escaped character| to false.
+
+1. Return [=success=] with data |result|.
+
+</div>
+
+<div algorithm>
+To <dfn>parse URL pattern</dfn>, given |pattern|:
+
+1. Let |has protocol| be true.
+
+1. Let |has hostname| be true.
+
+1. If |pattern| matches the <code>network.UrlPatternPattern</code> production:
+
+  1. Let |pattern url| be the empty string.
+
+    1. If |pattern| [=map/contains=] "<code>protocol</code>":
+
+      1. If |pattern|["<code>protocol</code>"] is the empty string, return [=error=]
+         with [=error code=] [=invalid argument=].
+
+      1. Let |protocol| be the result of [=trying=] to [=unescape
+         URL Pattern=] with |pattern|["<code>protocol</code>"].
+
+      1. For each |codepoint| in |protocol|:
+
+      1. If |codepoint| is not [=ASCII alphanumeric=] and «U+002B (+), U+002D
+         (-), U+002E (.)» does not [=set/contains|contain=] |codepoint|:
+
+        1. Return [=error=] with [=error code=] invalid argument.
+
+      1. Append |protocol| to |pattern url|.
+
+    1. Otherwise:
+
+      1. Set |has protocol| to false.
+
+      1. Append "<code>http</code>" to |pattern url|.
+
+    1. Let |scheme| be [=ASCII lowercase=] with |pattern url|.
+
+    1. Append "<code>:</code>" to |pattern url|.
+
+    1. If |scheme| [=is special=], append "<code>//</code>" to |pattern url|.
```

Comment:
For non special schemes (let's take `git` for instance), the URL will be parsed differently depending whether we add `//` or not:
- for `git://github.com/whatwg/url.git`, the host will be "github.com"
- for `git:github.com/whatwg/url.git` the host is empty

With the current API users can't really specify the first variant if they use URLPatternPattern. 
But maybe it doesn't matter? I imagine we are not actually interested in handling all possible URLs, but only whatever can be intercepted, so special schemes + a select few (eg `data:` ?). Do we need to surface this somehow in the spec?